src/main/java/org/spongepowered/api/resource/ResourceManager.java

```diff
@@ -0,0 +1,95 @@
+/*
+ * This file is part of SpongeAPI, licensed under the MIT License (MIT).
+ *
+ * Copyright (c) SpongePowered <https://www.spongepowered.org>
+ * Copyright (c) contributors
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package org.spongepowered.api.resource;
+
+import org.spongepowered.api.resource.pack.Pack;
+import org.spongepowered.api.resource.pack.PackInformationManager;
+import org.spongepowered.plugin.PluginContainer;
+
+import java.io.Closeable;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.util.Collection;
+import java.util.function.Predicate;
+import java.util.stream.Stream;
+
+/**
+ * The resource manager is in charge of loading {@link Resource Resources}.
+ *
+ * <p>Packs are stacked on top of each other, so they will override and replace
+ * resources in packs which are a lower priority.</p>
+ */
+public interface ResourceManager {
+
+    /**
+     * Gets the {@link PackInformationManager}.
+     *
+     * @return The pack information manager
+     */
+    PackInformationManager packInformationManager();
+
+    /**
+     * Loads the {@link Resource resource} at the given path, or throws an
+     * exception if it doesn't exist.
+     *
+     * <p>Resource implements {@link Closeable}, so remember to close it.</p>
+     *
+     * <p>Example:</p>
+     *
+     * <pre>
+     *     try (Resource res = resourceManager.load(path)) {
```

Comment:
For this to be true, `Resource` should actually implement `AutoClosable`.

...though do we really need to provide Java 102 in our javadocs?